lib/service/database.js

```diff
@@ -276,22 +277,37 @@ DatabaseService.prototype.queueAnalysisOperations = function(analysis, callback)
                 }
                 return _status;
             }, Node.STATUS.PENDING);
-            asyncQueries.push(updateNodeAtAnalysisCatalogQuery(nodeToUpdate.id(), status));
+            asyncQueries.push({
+                query: updateNodeAtAnalysisCatalogQuery(nodeToUpdate.id(), status),
+                onerror: updateNodeAtAnalysisCatalogQuery(nodeToUpdate.id(), Node.STATUS.FAILED)
+            });
             nodeToUpdate.setStatus(status);
         });
 
         nodesToQueueForUpdate.forEach(function(nodeToUpdate) {
             if (nodeToUpdate.shouldCacheQuery()) {
                 var targetTableName = nodeToUpdate.getTargetTable();
-                asyncQueries.push(deleteFromCacheTableQuery(targetTableName));
-                asyncQueries.push(updateNodeAtAnalysisCatalogQuery(nodeToUpdate.id(), Node.STATUS.RUNNING));
-                asyncQueries.push(populateCacheTableQuery(targetTableName, nodeToUpdate.sql()));
+                asyncQueries.push({
```

Comment:
duplicated code, but It's needed due to the loop goes through several nodes and we need to pass the specific node-id to the fallback.

I'd like explore other ways to do that.